### ISSUE TITLE
How to provide overloaded java functions to the jinjava template engine?

### ISSUE THREAD

**Issue by ruozhixian** [HubSpot/jinjava#320]
I want to provide 2 functions which have the same function name but different parameters:

`jinjava.getGlobalContext().registerFunction(new ELFunctionDefinition("", "tasks", Function.class, "tasks", String.class));`
`jinjava.getGlobalContext().registerFunction(new ELFunctionDefinition("", "tasks", Function.class, "tasks", String.class, boolean.class));`

to analyse the jinja sentences:

`{{ tasks(None) }}`
`{{ tasks(None, false) }}`

However, I find that the second "tasks" function would cover the first one, and there is only one function named "tasks“ in jinjava. So the first jinja sentence can't be solved.

If I want to use the 2 "tasks" functions, I have to use different "namespace" or "localName" when I add them to jinjava engine.

So, are there any method to add 2 or more overloaded java functions to jinjava and use them like "overloaded" with same function name in jinja?

**Comment by boulter** [comment]
You're right that you can't overload methods. I would just add some logic into one function that looks at the types of the parameters passed and calls different methods internally. We do that in various places to emulate varargs.
